tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala

```diff
@@ -0,0 +1,46 @@
+package dotty.tools.tasty
+
+import java.util.UUID
+
+import TastyFormat.{MajorVersion, MinorVersion, header}
+
+class TastyHeaderUnpickler(reader: TastyReader) {
+  import reader._
+
+  def this(bytes: Array[Byte]) = this(new TastyReader(bytes))
+
+  /**delegates to `readFullHeader`, expecting the following
+   * - `majorVersion == TastyFormat.MajorVersion`
+   * - `0 <= minorVersion <= TastyFormat.MinorVersion`
+   */
+  def readHeader(): UUID = {
+    val h = readFullHeader()
+    val validVersion = (
+      h.majorVersion == MajorVersion
+      && h.minorVersion >= 0 && h.minorVersion <= MinorVersion
+    )
+    check(validVersion,
+      s"""TASTy signature has wrong version.
+         | expected: $MajorVersion.$MinorVersion
+         | found   : ${h.majorVersion}.${h.minorVersion}""".stripMargin
+    )
```

Comment:
Is there any reason not to do this in `readFullHeader()`?

Reply:
I made an iteration where the user passed in a custom range to validate against, but then perhaps they may want to apply custom logic not anticipated

Reply:
> but then perhaps they may want to apply custom logic not anticipated

Don't prepare for completely unknown use cases. The use cases we know about will always want the validation, so we should validate. In the unlikely event that someone wants a non-validating variant, we can add another method in the future whose name clearly says that it doesn't validate.

Reply:
should the version to validate against be passed as a parameter?

Reply:
I don't think so. Simply do as `readHeader()` currently does.

Reply:
done